### PR TITLE
Arsenal - 3DEN Attribute

### PR DESCRIPTION
**Pull request by mharis001**
**When merged this pull request will:**
- Add a 3DEN attribute to allow adding a customized ACE Arsenal to objects
arsenal_attribute_4

### CODE REVIEW

**Comment by CPTMiller** [comment]
Would it be possible to add Insignia, Face and Voice to this?

**Comment by mharis001** [comment]
Those 3 categories do not go through the usual filtering. If enabled (through `ace_arsenal_enableIdentityTabs`) all insignia, faces, and voices are shown/available in arsenal.

**Comment by CPTMiller** [comment]
Oh I see. Yea I just want to be able to disable identity tabs.

**Comment by alganthe** [comment]
^ Those are disabled when using ACE Arsenal in 3DEN already.

**Comment by CPTMiller** [comment]
I was hoping to be able to disable them entirely, regardless if its singleplayer, multiplayer, dedicated or 3DEN. Most communities wants to be able to restrict its playerbase with whitelisting of gear and also faces, voices or insignia. Either if they want them entirely disabled or restricted down to which is allowed or not.

**Comment by alganthe** [comment]
I also added a 3DEN setting to disable them, you can also disable them manually by using the displayOpened EH

File: addons/arsenal/functions/fnc_attributeDblClick.sqf [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ */
+#include "script_component.hpp"
+#include "..\defines.hpp"
+
+params ["_listbox", "_currentRow"];
+TRACE_1("Double click toggle",_currentRow);
+
+// Get toggle mode (add or remove item)
+private _itemClassname = _listbox lnbData [_currentRow, 1];
+private _addItem = [true, false] select (_itemClassname in ((uiNamespace getVariable [QGVAR(attributeValue), [[], 0]]) select 0));
```
Comment (alganthe) [inline addons/arsenal/functions/fnc_attributeDblClick.sqf]: wut ???

File: addons/arsenal/functions/fnc_attributeDblClick.sqf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [CONTROL, 0] call ace_arsenal_fnc_attributeDblClick
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+#include "..\defines.hpp"
```
Comment (alganthe) [inline addons/arsenal/functions/fnc_attributeDblClick.sqf]: Including defines not needed here.

File: addons/arsenal/functions/fnc_attributeInit.sqf [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [box, [[], 1]] call ace_arsenal_fnc_attributeInit
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+#include "..\defines.hpp"
```
Comment (alganthe) [inline addons/arsenal/functions/fnc_attributeInit.sqf]: Including defines not needed here.

File: addons/arsenal/functions/fnc_attributeLoad.sqf [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [CONTROL, [[], 0]] call ace_arsenal_fnc_attributeLoad
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+#include "..\defines.hpp"
```
Comment (alganthe) [inline addons/arsenal/functions/fnc_attributeLoad.sqf]: Including defines not needed here.

**Comment by alganthe** [comment]
looks good to go, there's just some minor nitpicks from me :D

File: addons/arsenal/functions/fnc_attributeDblClick.sqf (hunk 1 above)
Comment (dedmen) [reply addons/arsenal/functions/fnc_attributeDblClick.sqf]: magic.

Comment (mharis001) [reply addons/arsenal/functions/fnc_attributeDblClick.sqf]: brain lag...

File: addons/arsenal/functions/fnc_attributeLoad.sqf (hunk 4 above)
Comment (mharis001) [reply addons/arsenal/functions/fnc_attributeLoad.sqf]: Needed here, IDCs used

Comment (alganthe) [reply addons/arsenal/functions/fnc_attributeLoad.sqf]: oh, right :D

File: addons/arsenal/Cfg3DEN.hpp [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+                collapsed = 1;
+                class Attributes {
+                    class ADDON {
+                        property = QGVAR(attribute);
+                        control = QGVAR(attribute);
+                        displayName = CSTRING(Mission);
+                        tooltip = "";
+                        expression = QUOTE(if (!is3DEN) then {[ARR_2(_this,+_value)] call FUNC(attributeInit)});
+                        defaultValue = "[[], 0]";
+                        condition = "1";
```
Comment (PabstMirror) [inline addons/arsenal/Cfg3DEN.hpp]: You can add it to soldiers, which works for accessing from other units interaction, but you can't access yourself.
Should we limit to just containers `objectHasInventoryCargo`
or just not soldiers `1 - objectControllable`
or add a self interaction action when added to CaManBase
or maybe it's not really a problem?

Comment (mharis001) [reply addons/arsenal/Cfg3DEN.hpp]: I think `1 - objectControllable` would be the best option.

File: addons/arsenal/functions/fnc_attributeAddItems.sqf [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [CONTROL, 0] call ace_arsenal_fnc_attributeAddItems
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
```
Comment (dedmen) [inline addons/arsenal/functions/fnc_attributeAddItems.sqf]: Need to move these. Either now. Or after merge in #6407

**Pull request merged**